### PR TITLE
feat: Allow indicative RFQ-T quotes without takerAddress

### PR DESCRIPTION
**Pull request by feuGeneA**
# Description

Changed RFQ-T enablement constraints such that a `takerAddress` query parameter is no longer required when coming through the `/price` endpoint.

# Testing Instructions

Use [Matcha on staging](https://api-staging.matcha.xyz/markets/DAI/USDC), and don't connect your wallet, and request some pricing, and check the API staging logs to ensure that RFQ-T liquidity was accessed.

# Checklist

-   [x] Implement.
-   [x] Test manually on staging.
-   [x] Merge [monorepo PR](https://github.com/0xProject/0x-monorepo/pull/2684) and re-pin the dependency herein to the `development` branch.
-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**

### CODE REVIEW

**Comment by feuGeneA** [comment]
deploy staging

**Comment by feuGeneA** [comment]
deploy staging

**Comment by feuGeneA** [comment]
deploy staging

**Comment by feuGeneA** [comment]
deploy staging

**Comment by feuGeneA** [comment]
deploy staging

File: src/handlers/swap_handlers.ts [hunk 1]
```diff
@@ -403,14 +403,24 @@ const parseGetSwapQuoteRequestParams = (
...
-        takerAddress && apiKey
-            ? {
-                  intentOnFilling: endpoint === 'quote' && req.query.intentOnFilling === 'true',
-                  isIndicative: endpoint === 'price',
-                  nativeExclusivelyRFQT,
-              }
-            : undefined;
+    const rfqt: Pick<RfqtRequestOpts, 'intentOnFilling' | 'isIndicative' | 'nativeExclusivelyRFQT'> = (() => {
+        if (apiKey) {
+            if (takerAddress) {
```
Comment (PirosB3) [inline src/handlers/swap_handlers.ts:408]: could you make the `if` statement the following? (just a suggestion)

```
if (endpoint === 'quote' && takerAddress) {
                return {
                    intentOnFilling: req.query.intentOnFilling === 'true',
                    isIndicative: false,
                    nativeExclusivelyRFQT,
                };
} else if (endpoint === 'price') {
                return {
                    intentOnFilling: false,
                    isIndicative: true,
                    nativeExclusivelyRFQT,
                };
} else {
    return undefined;
}
```

File: src/services/swap_service.ts [hunk 2]
```diff
@@ -471,7 +471,7 @@ export class SwapService {
                 },
             ]);
         }
-        if (apiKey !== undefined && (isETHSell || from !== undefined)) {
+        if (apiKey !== undefined && (isETHSell || from !== undefined || (rfqt && rfqt.isIndicative))) {
```
Comment (PirosB3) [inline src/services/swap_service.ts:474]: Can you explain this flow to me? why is this change needed?

Comment (feuGeneA) [reply src/services/swap_service.ts:474]: This conditional says, "if there's an API key AND (if either it's a forwarder tx (`isETHSell`) OR there's a taker address (`from`) OR it's an indicative quote".

Without this addition, an indicative quote request without a taker address cannot flow through here, so `_rfqt` remains undefined, so RFQT never gets enabled downstream.

File: src/handlers/swap_handlers.ts (hunk 1 above)
Comment (feuGeneA) [reply src/handlers/swap_handlers.ts]: Great suggestion, thank you.  Addressed in https://github.com/0xProject/0x-api/pull/320/commits/e63ee3b02ede33bd30f566ae8db6825d296523d9

File: package.json [hunk 3]
```diff
@@ -126,7 +126,7 @@
     },
     "dependencies": {
         "@0x/assert": "^3.0.4",
-        "@0x/asset-swapper": "0xProject/gitpkg-registry#0x-asset-swapper-v4.6.0-a2f0d5eed",
+        "@0x/asset-swapper": "0xProject/gitpkg-registry#0x-asset-swapper-v4.6.0-db0f3a878",
```
Comment (PirosB3) [inline package.json:129]: Should this contain the hash of the latest `master` when https://github.com/0xProject/0x-monorepo/pull/2684#pullrequestreview-478852431 is merged?

File: src/services/swap_service.ts (hunk 2 above)
Comment (PirosB3) [reply src/services/swap_service.ts:479]: This now makes sense for me, however, I'm a little concerned that whoever next touches this code won't understand the logic. Is it worth adding some comments explaining the reasoning behind L479? alternatively, we could refactor the `if` statement to use self-explaining variablenames. Example:

```
if (hasAPIKey && (isETHSell || hasTakerAddress || isIndicativeQuote)) {

}
```

File: package.json (hunk 3 above)
Comment (feuGeneA) [reply package.json]: Yup, done in https://github.com/0xProject/0x-api/pull/320/commits/c38fca77f088a4dbf328d17c194e02dffbb265ea

**Comment by feuGeneA** [comment]
deploy staging
